### PR TITLE
rev

### PR DESCRIPTION
**Pull request by MikeJSeo**
Here's an attempt.. 
@gravesti @miranta-antoniou-roche

### CODE REVIEW

File: R/matching.R [hunk 1]
```diff
@@ -147,6 +151,114 @@ plot_weights <- function(wt, bin_col = "#6ECEB2", vline_col = "#688CE8", main_ti
...
+    if (is.null(bin_col)) bin_col <- "black"
+    if (is.null(vline_col)) vline_col <- "red"
+  }
+
+  if (!ggplot) {
+    plot_weights_base(x, bin_col, vline_col, main_title, scaled_weights)
+  } else {
+    plot_weights_ggplot(x, bin_col, vline_col, main_title, bins, print_caption, caption_width)
+  }
+}
```
Comment (gravesti) [inline R/matching.R:259]: You could combine this into one `if (isTRUE(ggplot)){ ... } else { ... }`

File: R/matching.R [hunk 2]
```diff
@@ -147,6 +151,114 @@ plot_weights <- function(wt, bin_col = "#6ECEB2", vline_col = "#688CE8", main_ti
...
+
+plot.maicplus_estimate_weights <- function(x, ggplot = FALSE,
+                                           bin_col = NULL, vline_col = NULL,
+                                           main_title = NULL, scaled_weights = TRUE,
+                                           bins = 50, print_caption = FALSE,
+                                           caption_width = 80) {
+  if (!ggplot) {
+    if (is.null(main_title)) main_title <- "Scaled Individual Weights"
+    if (is.null(bin_col)) bin_col <- "#6ECEB2"
+    if (is.null(vline_col)) vline_col <- "#688CE8"
```
Comment (gravesti) [inline R/matching.R:247]: Is there a reason for different colours for base/ggplot?

**Review by gravesti — COMMENTED** [review]
Thanks @MikeJSeo 
I've made some suggestions. See if you can integrate the footnote text function we just merged :)

File: R/matching.R [hunk 3]
```diff
@@ -147,6 +151,114 @@ plot_weights <- function(wt, bin_col = "#6ECEB2", vline_col = "#688CE8", main_ti
...
+  wt_data <- stack(wt_data0)
+  wt_data$median <- ifelse(wt_data$ind == main_title[1], median(wt_data0[, main_title[1]]), median(wt_data0[, main_title[2]]))
+
+  summ <- aggregate(wt_data$values, list(wt_data$ind), median)
+  colnames(summ) <- c("ind", "lab")
+  summ$lab <- paste0(
+    "Median = ", round(summ$lab, 4),
+    "\nESS = ", round(weighted_data$ess, 2),
+    "\nReduction% = ", round((1 - (weighted_data$ess / dim(weighted_data$data)[1])) * 100, 2)
+  )
```
[the remark this answers is not in the document]
Comment (MikeJSeo) [reply R/matching.R:194]: Yea put in function called calculate_weights_legend

File: R/matching.R (hunk 1 above)
Comment (MikeJSeo) [reply R/matching.R]: combined

File: R/matching.R [hunk 4]
```diff
@@ -147,6 +151,114 @@ plot_weights <- function(wt, bin_col = "#6ECEB2", vline_col = "#688CE8", main_ti
...
+#' @param ggplot indicator to print base weights plot or ggplot weights plot
+#' @param bin_col a string, color for the bins of histogram
+#' @param vline_col a string, color for the vertical line in the histogram
+#' @param main_title title of the plot. For ggplot, name of scaled weights plot and unscaled weights plot, respectively.
+#' @param scaled_weights (base plot only) an indicator for using scaled weights instead of regular weights
+#' @param bins (ggplot only) number of bin parameter to use
+#' @param print_caption (ggplot only) print a footnote message related to ESS from the NICE survey 2021
+#' @param caption_width (ggplot only) width that is passed onto str_wrap function
+#'
+#' @examples
```
[the remark this answers is not in the document]
Comment (MikeJSeo) [reply R/matching.R:241]: added example!

File: R/matching.R [hunk 5]
```diff
@@ -147,6 +151,114 @@ plot_weights <- function(wt, bin_col = "#6ECEB2", vline_col = "#688CE8", main_ti
...
+    ggplot2::theme(
+      axis.title = ggplot2::element_text(size = 16),
+      axis.text = ggplot2::element_text(size = 16)
+    ) +
+    ggplot2::ylab("Frequency") +
+    ggplot2::xlab("Weight")
+
+  if (print_caption == TRUE) {
+    print_text <- "In most applications, weighting considerably reduces the effective sample size from the original AC sample size. The median percentage reduction is 58% (range: 7.9%–94.1%; interquartile range: 42.2%–74.2%). The final effective sample sizes are also representative of those in the technology appraisals, which are also small (median: 80; range: 4.8–639; interquartile range: 37–174). Therefore, an ESS reduction up to ~60% is not unexpected based on the 2021 survey, whereas a reduction of >75% is less common and it may be considered suboptimal."
+    hist_plot <- hist_plot + ggplot2::labs(caption = stringr::str_wrap(print_text, width = caption_width)) +
```
[the remark this answers is not in the document]
Comment (MikeJSeo) [reply R/matching.R]: I used ess_footnote_text

**Comment by MikeJSeo** [comment]
Hi guys, made some modifications!
@gravesti @miranta-antoniou-roche

File: R/matching.R [hunk 6]
```diff
@@ -145,8 +182,128 @@ plot_weights <- function(wt, bin_col = "#6ECEB2", vline_col = "#688CE8", main_ti
   axis(2, las = 1)
   abline(v = median(wt), lty = 2, col = vline_col, lwd = 2)
   legend("topright", bty = "n", lty = plot_lty, cex = 0.8, legend = plot_legend)
+  if (print_caption) {
+    text <- ess_footnote_text(width = caption_width)
+    mtext(text, adj = 0, side = 1, line = str_count(text, "\n") + 1, font = 3, cex = 0.7, col = blues9)
```
Comment (gravesti) [inline R/matching.R:187]: we can take this out and put this in a wrapper with layout or similar.

**Review by gravesti — APPROVED** [review]
@MikeJSeo I think we can merge this now :)

**Pull request merged**